packages/persist/lib/records.ts

```diff
@@ -59,6 +59,16 @@ export async function persistRecords({
         }
     });
 
+    const logCtx = logContextGetter.getStateLess({ id: String(activityLogId), accountId });
+
+    const connection = await connectionService.getConnectionById(nangoConnectionId);
+    if (!connection) {
+        const err = new Error(`Connection ${nangoConnectionId} not found`);
+        void logCtx.error('Connection not found', { error: err, persistType });
+        span.setTag('error', err).finish();
+        return Err(err);
+    }
+
```

Comment:
getting the connection is purely internal logic. I don't think we should expose that to the user, or even fail.